visualization/app/codeCharta/ui/sortingButton/sortingButton.component.ts

```diff
@@ -1,34 +1,18 @@
-import "./sortingButton.component.scss"
-import { IRootScopeService } from "angular"
-import { StoreService } from "../../state/store.service"
+import { Component } from "@angular/core"
+
+import { connect } from "../../state/angular-redux/connect"
 import { setSortingOrderAscending } from "../../state/store/appSettings/sortingOrderAscending/sortingOrderAscending.actions"
-import {
-	SortingOrderAscendingService,
-	SortingOrderAscendingSubscriber
-} from "../../state/store/appSettings/sortingOrderAscending/sortingOrderAscending.service"
+import { sortingOrderAscendingSelector } from "../../state/store/appSettings/sortingOrderAscending/sortingOrderAscending.selector"
+import { CcState } from "../../state/store/store"
 
-export class SortingButtonController implements SortingOrderAscendingSubscriber {
-	private _viewModel: {
-		orderAscending: boolean
-	} = {
-		orderAscending: true
-	}
-	constructor(private $rootScope: IRootScopeService, private storeService: StoreService) {
-		"ngInject"
-		SortingOrderAscendingService.subscribe(this.$rootScope, this)
-	}
-
-	onSortingOrderAscendingChanged(sortingOrderAscending: boolean) {
-		this._viewModel.orderAscending = sortingOrderAscending
-	}
+const ConnectedClass = connect((state: CcState) => ({ orderAscending: sortingOrderAscendingSelector(state) }), { setSortingOrderAscending })
 
+@Component({
+	selector: "cc-sorting-button",
```

Comment:
Why the `cc-` prefix? use `sorting-button` or `sorting-button-component` instead

Reply:
if I recall correctly product / company prefix is best practice for custom elements. They have to be unique and it "avoids" name clashes in case of multiple custom elements from different providers. It is also less verbose than component and ensures that each component has a dash, which is required.